### PR TITLE
fix: mongo replica set

### PR DESCRIPTION
**Pull request by klassm**
From typeorm > 0.2.28 parsing the connection string broke our database setup. In our setup we use a replicaSetup - however the required parameters tls and replicaSet were never added to the connection string.

This PR will add the two parameters, in addition to a new test that parses the connection string (as I was looking for how the parsing works).

There is no new test for the actual adding of the two query parameters. If you can point me towards an example test case, I can also provide one for this case.

### Pull-Request Checklist

- [x] Code is up-to-date with the `master` branch
- [x] `npm run lint` passes with this change
- [x] `npm run test` passes with this change
- [x] This pull request links relevant issues as `Fixes #0000`
- [x] There are new or updated unit tests validating the change
- [x] Documentation has been updated to reflect this change
- [x] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/typeorm/typeorm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: test/utils/test-utils.ts [hunk 1]
```diff
@@ -42,6 +42,11 @@ export interface TestingOptions {
      */
     name?: string;
 
+    /**
+     * Url for the database connection.
+     */
+    url?: string;
```
Comment (imnotjames) [inline test/utils/test-utils.ts:48]: Why this change?

Comment (klassm) [reply test/utils/test-utils.ts:48]: This change was more about being able to test if. If you want we can also revert it - but generally I tried it out by modifying the `ormconfig.json`, which required me to allow setting a url here.

Comment (imnotjames) [reply test/utils/test-utils.ts:48]: I get that it's for testing, but no test you added uses this.

We technically can't, either, because the mongo server we set up won't have many features related to multiple instances enabled.

Comment (klassm) [reply test/utils/test-utils.ts:48]: Yes exactly. If you want I can also remove it again :-)

**Comment by klassm** [comment]
I removed the corresponding commit, so the offending line is gone again :-)

File: test/utils/test-utils.ts (hunk 1 above)
Comment (klassm) [reply test/utils/test-utils.ts]: I removed the corresponding commit, so the offending line is gone again :-)

File: src/driver/mongodb/MongoDriver.ts [hunk 2]
```diff
@@ -442,21 +442,22 @@ export class MongoDriver implements Driver {
     /**
      * Builds connection url that is passed to underlying driver to perform connection to the mongodb database.
      */
-    protected buildConnectionUrl(options: { [key: string]: any }): string {
+    static buildConnectionUrl(options: { [key: string]: any }): string {
```
Comment (imnotjames) [inline src/driver/mongodb/MongoDriver.ts:445]: Can this still be protected even if it's static?  I guess not because that's how you're confirming this with tests?

Is there any way to test that this change works WITHOUT exposing more functions as part of a public API?

Comment (klassm) [reply src/driver/mongodb/MongoDriver.ts:445]: Yeah we thought about this two. The unfortunate part is that the one public method using this is the connect method - so then we'd have to mock the world to get a test. But I wanted to have a test 🙈

Comment (klassm) [reply src/driver/mongodb/MongoDriver.ts]: Ok now I rewrote the test a bit - don't know whether it's really pretty now. But what it does now is mock the connect function and emulate what the mongodriver does, so we can actually catch the connection url and verify that it is correct. A bit hard to read, but that way there is no additional public method.

**Review by imnotjames — APPROVED** [review]
